### DISCUSSION TITLE
Intellectual property ownership

### DISCUSSION THREAD

**Discussion by rbamini** [Budibase/budibase#3535] [General]
Hi all - wondering if anyone knows more about ownership rights for apps created with Budibase's free option. Obviously not expecting legal advice here, but the below from the Terms of Service got me wondering if there's a difference between the platform's code being open-source versus their control/ownership of anything created with the platform. Keeping in mind I also know almost nothing about software development, so if there's an obvious technical reason why they can't track/control what's made with their platform, feel free to point that out too :)

_9.1 Customer acknowledges and agrees that BB and/or its licensors own all intellectual property rights in, or arising from, the Services, Software and their documentation. All content on BB’s site and the Subscription Service is copyright © Budibase Limited. The Budibase™ name and brand, as well as any other BB trademarks (word marks, picture marks or integrated marks) and associated unregistered trademarks and trade names used by BB. Except as expressly stated herein, the Agreement does not grant Customer any rights to, or in, patents, copyrights, database rights, trade secrets, trade names, trade marks (whether registered or unregistered), or any other rights or licences in respect of the Services, Software, their documentation and the BB trademark or associated unregistered trademarks. This shall be without prejudice to Customer’s ownership of any background intellectual property rights owned by it separately from the Agreement and/or pre-dating the Agreement._

**Comment by manojahi** [comment]
Great question, I was also wondering about the copyright. I would assume that `If i am hosting on my own cloud, then i should be the owner of the property`.

but looking forward to the other replies.

**Comment by mjashanks** [comment]
Hi There. 

We have 3 separate licenses for Budibase, used for different purposes.

### [1. Budibase cloud terms of service.](https://budibase.com/terms-of-service)

This only applies to those using Budibase on our cloud platform: https://*.budibase.app.

We definitely do not expect to own the IP to the apps that you build. However, I can see why this paragraph would seem to contradict that. I will arrange to have this amended, to clarify that! So thanks for pointing that out.

You also mention open-source here. Budibase is indeed open-source. However, open-source only applies when you are self-hosting. Our cloud platform runs the open-source product but is not open-source licensed in itself. I.e. we have additional code and infrastructure on top of the open-source product that allows us to operate our cloud environment.

Our Cloud license also comes with additional restrictions on what you are allowed to use it for - e.g. nothing illegal! So, this is another one of the many reasons why our cloud platform is licensed separately. Our open-source license has no such restriction.

### [2. Budibase open source (self-host)](https://github.com/Budibase/budibase/blob/master/LICENSE)

When you self-host the open-source product, the only terms that apply to you are our open-source license - GPLv3. In this case, all apps that you build belong to you, are your responsibility and can be licensed in any way you see fit. 

No other terms apply to you in this scenario.

### [3. Budibase self-host commercial license](https://budibase.com/self-host-master-terms)

This license is for customers who want to
- Self-host
- Pay us for extended liability 
- Pay us for services or premium feature

It's a commercial license, made for those running Budibase for critical applications and therefore need us (Budibase Ltd. the company) to be there to support them.

Again, we have no intention of claiming IP on apps built under this license.

**Reply by krkeegan** [reply]
> all apps that you build belong to you, are your responsibility and can be licensed in any way you see fit.

That is a bit misleading, the client apps at minimum are licensed under the Mozilla Public License which in Section 3.2 requires any distribution to make the Source Code available.

**Reply by mjashanks** [reply]
Yea - fair point. However, budibase apps cannot be ejected from the Budibase platform, so I'm not sure how worthwhile it is to modify the source of the outputted apps. They always rely on the whole budibase stack.

My point was that we don't claim to own the apps that you build with Budibase. I'm not against making the client libraries MIT licensed if it's something that matters to people. The overall Budibase stack will remain GPL.

**Reply by krkeegan** [reply]
Well, it would be a violation of the license not to make the source code and license available, I note it must be readable source and not minified or obfuscated.  So user should know that in order to comply with the license, they need to make these things available, even if from a technical perspective it doesn't make much sense.

In addition, commercial clients are always concerned about copycats.  The obligation to provide the client app source code could give competitors a mild advantage in duplicating the work.

I realize that even without the obligation to distribute the source code, that javascript and html are still human readable to an extent.  I also understand that the backend functionality, which is not distributed and would not require source code distribution, is still critically important and just taking the client app source code available would not immediately allow for easy copying.

If you are open to MIT licensing the client libraries, that would make things much easier for users and then your statement that user can "licensed in any way you see fit" would be accurate.
